java-util/src/main/java/io/druid/java/util/common/granularity/Granularity.java

```diff
@@ -407,82 +410,15 @@ else if (vals[index] == 5) {
           else if (vals[index] == 1) {
             return GranularityType.MINUTE;
           }
-        }
-        if (index == 6) {
+          break;
+        case 6:
           return GranularityType.SECOND;
-        }
+        default:
+          break;
       }
-
       throw new IAE("Granularity is not supported. [%s]", period);
     }
 
-    static DateTime getDateTime(GranularityType gran, Integer[] vals)
-    {
-      switch (gran) {
-        case SECOND: {
-          DateTime date = null;
-          if (vals[1] != null
-              && vals[2] != null
-              && vals[3] != null
-              && vals[4] != null
-              && vals[5] != null
-              && vals[6] != null) {
-            date = new DateTime(vals[1], vals[2], vals[3], vals[4], vals[5], vals[6], 0);
-          }
-
-          return date;
-        }
-        case MINUTE:
-        case FIVE_MINUTE:
-        case TEN_MINUTE:
-        case FIFTEEN_MINUTE:
-        case THIRTY_MINUTE: {
-          DateTime date = null;
-          if (vals[1] != null && vals[2] != null && vals[3] != null && vals[4] != null && vals[5] != null) {
-            date = new DateTime(vals[1], vals[2], vals[3], vals[4], vals[5], 0, 0);
-          }
-          return date;
-        }
-        case HOUR:
-        case SIX_HOUR: {
-          DateTime date = null;
-          if (vals[1] != null && vals[2] != null && vals[3] != null && vals[4] != null) {
-            date = new DateTime(vals[1], vals[2], vals[3], vals[4], 0, 0, 0);
-          }
-
-          return date;
-        }
-        case DAY:
-        case WEEK: {
-          DateTime date = null;
-          if (vals[1] != null && vals[2] != null && vals[3] != null) {
-            date = new DateTime(vals[1], vals[2], vals[3], 0, 0, 0, 0);
-          }
-
-          return date;
-        }
-        case MONTH:
-        case QUARTER: {
-          DateTime date = null;
-          if (vals[1] != null && vals[2] != null) {
-            date = new DateTime(vals[1], vals[2], 1, 0, 0, 0, 0);
-          }
-
-          return date;
-        }
-        case YEAR: {
-          DateTime date = null;
-          if (vals[1] != null) {
-            date = new DateTime(vals[1], 1, 1, 0, 0, 0, 0);
-          }
-
-          return date;
-        }
-      }
-
-      return null;
-    }
-
     public String getHiveFormat()
     {
       return hiveFormat;
```

Comment:
Below in this class, `IntervalIterable` and `IntervalIterator`, and their constructors could be private.